addons/misra.py

```diff
@@ -808,17 +808,29 @@ def misra_10_8(data):
             continue
         if not token.astOperand1.astOperand1:
             continue
-        try:
-            intTypes = ['char', 'short', 'int', 'long', 'long long']
-            index1 = intTypes.index(token.valueType.type)
-            e = getEssentialType(token.astOperand1)
-            if not e:
+        if token.astOperand1.str not in {'+', '-', '*', '/', '%', '&', '|', '^', '>>', "<<", "?", ":", '~'}:
```

Comment:
Should there be 10.8 warnings for unary operators? otherwise it seems to me we should continue for unary operators.